bundles/org.openhab.core.crypto.cryptostore/src/main/java/org/openhab/core/crypto/cryptostore/CryptoStore.java

```diff
@@ -0,0 +1,328 @@
+        setPrivKey(privKey, keyString);
+        setCert(cert);
+    }
+
+    public void setKeyStore(String keystoreFileName) {
+        this.keystoreFileName = keystoreFileName;
+    }
+
+    public void loadFromKeyStore(String keystoreFileName, String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        this.keystoreFileName = keystoreFileName;
+        loadFromKeyStore(keystorePassword, keyString);
+    }
+
+    public void loadFromKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        Key key = convertByteToKey(keyString);
+        KeyStore keystore = KeyStore.getInstance("JKS");
+        FileInputStream keystoreInputStream = new FileInputStream(this.keystoreFileName);
+        keystore.load(keystoreInputStream, keystorePassword.toCharArray());
+        byte[] byteKey = keystore.getKey(this.alias, keystorePassword.toCharArray()).getEncoded();
+        this.privKey = encrypt(new String(Base64.getEncoder().encode(byteKey)), key);
+        setCert(keystore.getCertificate(this.alias));
+        String caAlias = this.caAlias;
+        Certificate caCert = keystore.getCertificate(caAlias);
+        if (caCert != null) {
+            setCaCert(caCert);
+        }
+    }
+
+    public KeyStore getKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        KeyStore keystore = KeyStore.getInstance("JKS");
+        keystore.load(null, null);
+        byte[] pkcs8EncodedBytes = Base64.getDecoder().decode(getPrivKey(keyString));
+        PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(pkcs8EncodedBytes);
+        KeyFactory kf = KeyFactory.getInstance(this.keystoreAlgorithm);
+        keystore.setKeyEntry(this.alias, kf.generatePrivate(keySpec), keystorePassword.toCharArray(),
+                new java.security.cert.Certificate[] { getCert() });
+        if (!caCert.isEmpty()) {
+            String caAlias = this.caAlias;
+            keystore.setCertificateEntry(caAlias, getCaCert());
+        }
+        return keystore;
+    }
+
+    public void saveKeyStore(String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        saveKeyStore(this.keystoreFileName, keystorePassword, keyString);
+    }
+
+    public void saveKeyStore(String keystoreFileName, String keystorePassword, byte[] keyString)
+            throws GeneralSecurityException, IOException, Exception {
+        FileOutputStream keystoreStream = new FileOutputStream(keystoreFileName);
+        KeyStore keystore = getKeyStore(keystorePassword, keyString);
+        keystore.store(keystoreStream, keystorePassword.toCharArray());
+    }
+
+    private X509Certificate generateSelfSignedCertificate(KeyPair keyPair, String distName)
```

Comment:
Why is `distName` passed as an argument instead of using the instance variable?

Reply:
I suppose I could pull that directly.  It's only called once internally and when it's called it uses this.distName.